cassandra-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/CassandraCluster.java

```diff
@@ -977,4 +1039,30 @@ private boolean isLiveNode(CassandraNode node) {
             return false;
         return info.getNativeTransportRunning() && info.getRpcServerRunning();
     }
+
+    public CassandraNode replaceNode(String node) throws ReplaceNodePreconditionFailed {
```

Comment:
Please add tests that check both the positive & negative cases of each of these branches. There is a lot of logic here that could go wrong if we don't account for something correctly.